client/src/components/MultiJurisdictionAudit/useCSV.ts

```diff
@@ -63,24 +79,44 @@ const useCSV = (
   () => Promise<boolean>
 ] => {
   const [csv, setCSV] = useState<IFileInfo | null>(null)
-
+  const [upload, setUpload] = useState<IUpload | null>(null)
+
+  // Load (or reload) this file whenever a file it depends on changes. This is
```

Comment:
Had to refine this logic because it wasn't working correctly once I added the upload progress to the hook state. Most likely the issue was that we were using an object (`dependencyFile`) in the useEffect dependency list, which is not very reliable because the objects are checked for changes using reference equality, not structural equality. By adding the upload progress to these objects, it changed something about the way this useEffect hook noticed changes, and started causing issues.

The fix was to pull out the specific pieces of info from the object that we wanted to check.